internal/declarative/v2/reconciler.go

```diff
@@ -155,10 +157,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.ssaStatus(ctx, obj)
 	}
 
-	if !obj.GetDeletionTimestamp().IsZero() {
+	resourceCRDeleted, err := r.DeletionCheck.Run(ctx, clnt, obj)
```

Comment:
why we need deletioncheck here? to verify resourceCR get deleted before remove manifest CR finalizer, it should and already detect in `doPreDelete`

Reply:
ok, we refactored the `doPreDelete` to not wait for CR deletion.

Reply:
This refactoring is not needed, the existing code just remove default finalizer `"declarative.kyma-project.io/finalizer"`, and in `doPreDelete` (`PreDeleteDeleteCR`), it will remove `"resource.kyma-project.io/finalizer"` after resourceCR and CRD get deleted. Which means manifest CR anyway will not deleted if that `"resource.kyma-project.io/finalizer"` still exists.

you current refactoring makes the manifest CR deletion logic complicated.